pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -293,10 +293,18 @@ func (at *ActivityTree) insertEvent(event *model.Event, dryRun bool, generationT
 		return false, fmt.Errorf("invalid event: %s", err)
 	}
 
-	node, newProcessNode, err := at.CreateProcessNode(event.ProcessCacheEntry, generationType, dryRun, resolvers)
+	createProcessNodeDryRun := dryRun
+	if event.GetEventType() != model.ExecEventType {
```

Comment:
Should we add a test to check if the `exec` event type is no longer in "learning" mode ?

Reply:
This logic is done upstream profile side (in TryAutolearn()), prior to the Insert() call. I think it's OK as this: keeping the profile logic in the profile and stop inserting dns/file/bind processes on activity trees.

Reply:
I see what you mean but let's chat offline, I actually believe we _**need**_ to insert processes based on DNS events because of use cases like the Datadog Agent !

Reply:
I just pushed a new version with the modification we agreed on! It should be fine now. Thanks for pointing me the issue.